validator/specs/lib/validateCmptSpec.js

```diff
@@ -0,0 +1,495 @@
+        expect(validateCmpt(createCmptExtraBytes()).result).toBe(false);
+    });
+
+    it('validates a cmpt tile with no inner tiles', function() {
+        expect(validateCmpt(createEmptyCmpt()).result).toBe(true);
+    });
+
+    it('validates a cmpt tile with a valid b3dm inner tile', function() {
+        expect(validateCmpt(createCmptB3dm()).result).toBe(true);
+    });
+
+    it('validates a cmpt tile with a valid i3dm inner tile', function() {
+        expect(validateCmpt(createCmptI3dm()).result).toBe(true);
+    });
+
+    it('validates a cmpt tile with a valid pnts inner tile', function() {
+        expect(validateCmpt(createCmptPnts()).result).toBe(true);
+    });
+
+    it('validates a cmpt tile with a valid combinations of inner tiles', function() {
+        expect(validateCmpt(createCmptCombination()).result).toBe(true);
+    });
+
+    it('returns false if the cmpt has an invalid b3dm inner tile', function() {
+        expect(validateCmpt(createCmptInvalidB3dm()).result).toBe(false);
+    });
+
+    it('returns false if the cmpt has an invalid i3dm inner tile', function() {
+        expect(validateCmpt(createCmptInvalidI3dm()).result).toBe(false);
+    });
+
+    it('returns false if the cmpt has an invalid pnts inner tile', function() {
+        expect(validateCmpt(createCmptInvalidPnts()).result).toBe(false);
+    });
+
+    it('returns false if the cmpt contains any invalid inner tiles', function() {
+        expect(validateCmpt(createCmptInvalidCombination()).result).toBe(false);
+    });
+});
+
+function createInvalidMagic() {
+    var cmptTile = new Buffer(16);
+    cmptTile.write('xxxx', 0); // magic
+    cmptTile.writeUInt32LE(1, 4); // version
+    cmptTile.writeUInt32LE(cmptTile.length, 8); // byteLength
+    cmptTile.writeUInt32LE(0, 12); // tilesLength
+
+    return cmptTile;
+}
+
+function createInvalidVersion() {
+    var cmptTile = new Buffer(16);
+    cmptTile.write('cmpt', 0); // magic
+    cmptTile.writeUInt32LE(5, 4); // version
+    cmptTile.writeUInt32LE(cmptTile.length, 8); // byteLength
+    cmptTile.writeUInt32LE(0, 12); // tilesLength
+
+    return cmptTile;
+}
+
+function createInvalidByteLength() {
+    var cmptTile = new Buffer(16);
+    cmptTile.write('cmpt', 0); // magic
+    cmptTile.writeUInt32LE(1, 4); // version
+    cmptTile.writeUInt32LE(cmptTile.length - 1, 8); // byteLength
+    cmptTile.writeUInt32LE(0, 12); // tilesLength
+
+    return cmptTile;
+}
+
+function createCmptShort() {
+    var byteOffset = 0;
+    var sizeU32Int = 4;
+    var cmptTile = new Buffer(48);
+
+    //cmpt header
+    cmptTile.write('cmpt', byteOffset); // magic
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(1, byteOffset); // version
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(cmptTile.length, byteOffset); // byteLength
+    byteOffset += sizeU32Int;
+    cmptTile.writeUInt32LE(5, byteOffset); // tilesLength
+    byteOffset += sizeU32Int;
```

Comment:
Instead of going with the `byteOffset` approach for building the test data, it may be worth considering `Buffer.concat` where the cmpt and i3dm and created separately and then merged (wherever this makes sense).